### PR TITLE
Add dev/preview banner

### PR DESCRIPTION
**Pull request by aapeliv**
Also enabled grpc web debugging in previews

Screen Shot 2021-04-07 at 10 29 49 am
Screen Shot 2021-04-07 at 10 29 40 am

### CODE REVIEW

**Comment by aapeliv** [comment]
Span hiding could be simplified with `useMediaQuery`

**Comment by aapeliv** [comment]
Tried doing it with a [Chip](https://material-ui.com/components/chips/) but can't make the color right

Screen Shot 2021-04-07 at 1 33 39 pm

File: app/frontend/src/components/EnvironmentBanner.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  },
+}));
+
+export function EnvironmentBanner() {
+  const theme = useTheme();
+  const classes = useStyles();
+  const isBelowSm = useMediaQuery(theme.breakpoints.down("sm"));
+
+  if (process.env.REACT_APP_COUCHERS_ENV === "prod") {
+    return <></>;
```
Comment (darrenvong) [inline app/frontend/src/components/EnvironmentBanner.tsx]:
```suggestion
    return null;
```
React convention is to return null when you don't want to anything

**Review by darrenvong — CHANGES_REQUESTED** [review]
This floating thing looks kinda ugly :p It might look a bit better to have it above the nav bar instead (similar to how the alpha site did it)?

**Comment by aapeliv** [comment]
> This floating thing looks kinda ugly :p It might look a bit better to have it above the nav bar instead (similar to how the alpha site did it)?

It's just that it then offsets other stuff and e.g. the logged out landing index page starts scrolling, etc?

**Comment by darrenvong** [comment]
> It's just that it then offsets other stuff and e.g. the logged out landing index page starts scrolling, etc?

True and it's probably gonna give false CSS positives in dev. How about having it on the bottom left next to where the React Query dev tools usually are?

**Comment by aapeliv** [comment]
Just needs to be prominent enough that people don't get confused (in particular if they end up there by accident)... We already had someone ask me to move their profile from the preview db to prod.

**Comment by lucaslcode** [comment]
Can we not merge this until #1003 is solved, there's a few work arounds I want to try

**Comment by aapeliv** [comment]
> Can we not merge this until #1003 is solved, there's a few work arounds I want to try

Sure

**Comment by lucaslcode** [comment]
Made it removeable, moved to bottom right, and made it pulse since I was worried about it not being seen down there

image

**Comment by aapeliv** [comment]
Does this now obstruct buttons that might appear on the bottom on a phone?

I had `pointerEvents: "none"` and a bit of transparency before so you could click through it, but obviously you can't do that if you can click it away. Or do you just click it away if you want to press a button there?

**Comment by darrenvong** [comment]
> Or do you just click it away if you want to press a button there?

I think that's ok since it will only be on dev builds?

**Review by darrenvong — APPROVED** [review]

**Pull request merged**